personalitytest/static/js/src/personalitytest-view.js

```diff
@@ -97,16 +103,15 @@ function PersonalityTestXBlockStudent(runtime, element) {
     getScore();
     initDisplay();
 
-    $('.personality-test-form').on('click', '.save-button', function () {
-        //e.preventDefault();
-        // addGroupForm.off('submit');
+    $('.personality-test-form').on('click', '.save-button', function (e) {
+        e.preventDefault();
 
         var json = '[';
         var first = true;
         var errors = 0;
         $('select option:selected', element).each(function () {
             var that = $(this);
-            if (that.text() === '-') {
+            if (that.val() === '') {
                 errors++;
```

Comment:
Also, there’s no need to loop through all the answers to just count the errors if just one is sufficient.

Break the loop if you find an error.